crates/matrix-sdk-base/src/event_cache/store/memory_store.rs

```diff
@@ -35,6 +36,7 @@ use crate::{
 pub struct MemoryStore {
     media: StdRwLock<RingBuffer<(OwnedMxcUri, String /* unique key */, Vec<u8>)>>,
     leases: StdRwLock<HashMap<String, (String, Instant)>>,
+    events: StdRwLock<RelationalLinkedChunk<Event, Gap>>,
```

Comment:
preexisting, but maybe an opportunity to rewrite it while we're around: could we use a single lock for all the data in that store? this would avoid lock ordering issues like we've had in the past with the other store. (So we could introduce a new `MemoryStoreInner`, and `MemoryStore` would hold an `StdRwLock<MemoryStoreInner>`.)

Reply:
Done.